agora-bench/Bench.hs

```diff
@@ -37,7 +37,11 @@ data Benchmark = Benchmark
   , scriptSize :: Int
   -- ^ The on-chain size of a script.
   }
-  deriving stock (Show, Eq, Ord, Generic)
+  deriving stock (Eq, Ord, Generic)
+
+instance Show Benchmark where
+  show (Benchmark name (ExCPU cpu) (ExMemory mem) size) =
+    unpack name <> "\n\tCPU: " <> show cpu <> " MEM: " <> show mem <> " SIZE: " <> show size
```

Comment:
While technically `Show` doesn't have laws, we should avoid pretty printing in `Show` instances. It will diverge from generated `Show` instances we're used to (this is why [this](https://github.com/haskell-text/text-display) exists). Instead, use the `Pretty` type class (we already have `prettyprinter` as a dep):
```haskell
instance Pretty Benchmark where
  pretty (Benchmark name (ExCPU cpu) (ExMemory mem) size) =
    vsep
      [ pretty name
      , indent 4 $ "CPU:" <+> pretty cpu <+> "MEM:" <+> pretty mem <+> "SIZE:" <+> pretty size
      ]
```

Then, you can `renderString (layoutPretty defaultLayoutOptions (pretty benchmark))`. It's a bit lengthy, but it does the job.